From 3902f1529ae277994eb669be5c078f0e1011d157 Mon Sep 17 00:00:00 2001
Message-ID: <3902f1529ae277994eb669be5c078f0e1011d157.1720090727.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1720090727.git.sharlatanus@gmail.com>
References: <cover.1720090727.git.sharlatanus@gmail.com>
From: Rodion Goritskov <rodion.goritskov@gmail.com>
Date: Wed, 3 Jul 2024 23:30:22 +0100
Subject: [PATCH v2 5/9] gnu: Add go-github-com-fxamacker-cbor-v2.

* gnu/packages/golang-xyz.scm (go-github-com-fxamacker-cbor-v2): New variable.

Signed-off-by: Sharlatan Hellseher <sharlatanus@gmail.com>
Change-Id: I19e13404586613fd4629c35f50264120df264c6b
---
 gnu/packages/golang-xyz.scm | 29 +++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index d63c184bf6..e7811766df 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -1775,6 +1775,35 @@ (define-public go-github-com-flynn-archive-go-shlex
       (home-page "https://github.com/flynn-archive/go-shlex")
       (license license:asl2.0))))
 
+(define-public go-github-com-fxamacker-cbor-v2
+  (package
+    (name "go-github-com-fxamacker-cbor-v2")
+    (version "2.7.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/fxamacker/cbor")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "039lk7n5155gy2sh55i1darcvxhv9fim2xmnvmx0xi9ihnrnczln"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:import-path "github.com/fxamacker/cbor/v2"))
+    (propagated-inputs
+     (list go-github-com-x448-float16))
+    (home-page "https://github.com/fxamacker/cbor")
+    (synopsis "CBOR Codec in Go")
+    (description
+     "This package implement functionality for encoding and decoding
+@acronym{Concise Binary Object Representation,CBOR}
+(@url{https://www.rfc-editor.org/rfc/rfc8949.html,RFC 8949}) and CBOR
+Sequences,with CBOR tags, Go struct tags (toarray, keyasint, omitempty),
+float64/32/16, big.Int.")
+    (license license:expat)))
+
 (define-public go-github-com-gabriel-vasile-mimetype
   (package
     (name "go-github-com-gabriel-vasile-mimetype")
-- 
2.41.0

